GUI/Types/Renderer/GLModelViewer.cs

```diff
@@ -331,6 +344,68 @@ protected override void OnPaint(object sender, RenderEventArgs e)
             }
 
             base.OnPaint(sender, e);
+
+            if (ModelStatsCheckbox.Checked)
+            {
+                selectedNodeRenderer.SelectNode(modelSceneNode);
+
+                using (new GLDebugGroup("Model Viewer Stats Text"))
+                {
+                    var meshesString = "";
+
+                    var modelMeshCount = 0;
+                    foreach (var embeddedMesh in model.GetEmbeddedMeshesAndLoD())
```

Comment:
This is horrible to call every draw, also doesn't work for separate mesh files. You should access the existing scene nodes.